### PR TITLE
Adds visual inspection of generated code and bug fixes

### PR DESCRIPTION
**Pull request by ericanderson**
This is not tied into `pnpm dev` so you will still have to run it by hand. We can add better support later.

### CODE REVIEW

File: .lintstagedrc.mjs [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default {
   "monorepo/**/*.{js,jsx,ts,tsx,mjs,cjs}": [
     "dprint fmt",
   ],
-  "{packages,examples}/**/*.{js,jsx,ts,tsx,mjs,cjs}": [
+  "{packages,examples/basic}/**/*.{js,jsx,ts,tsx,mjs,cjs}": [
```
Comment (ericanderson) [inline .lintstagedrc.mjs:18]: Ignoring formatting checks on the one_dot_one

File: examples/one_dot_one/src/generatedNoCheck/FoundryClient.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Path: /Volumes/git/public-osdk/examples/one_dot_one/src/generatedNoCheck/FoundryClient
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/FoundryClient.ts:1]: Nit, we can remove path.

File: examples/one_dot_one/src/generatedNoCheck/Ontology.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Path: /Volumes/git/public-osdk/examples/one_dot_one/src/generatedNoCheck/Ontology
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/Ontology.ts:1]: Nit: path

File: examples/one_dot_one/src/generatedNoCheck/index.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Path: /Volumes/git/public-osdk/examples/one_dot_one/src/generatedNoCheck/index.ts
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/index.ts:1]: Nit path

File: examples/one_dot_one/src/generatedNoCheck/objects/Person.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { ObjectDefinition } from '@osdk/api';
+
+import { OntologyObject, LocalDate, Timestamp, GeoShape, GeoPoint } from '@osdk/legacy-client';
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/objects/Person.ts:3]: Nit: imports should be types where possible

File: examples/one_dot_one/src/generatedNoCheck/objects/Todo.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { ObjectDefinition } from '@osdk/api';
+
+import { OntologyObject, LocalDate, Timestamp, GeoShape, GeoPoint } from '@osdk/legacy-client';
+/**
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/objects/Todo.ts:4]: nit add newline.

File: examples/one_dot_one/src/generatedNoCheck/ontologyObjects.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// Path: /Volumes/git/public-osdk/examples/one_dot_one/src/generatedNoCheck/ontologyObjects.ts
+import { BaseObjectSet } from '@osdk/legacy-client';
+import { Todo, Person } from './objects';
+export interface Objects {
```
Comment (ericanderson) [inline examples/one_dot_one/src/generatedNoCheck/ontologyObjects.ts:4]: nit new line

File: monorepo/eslint-config-sane/library.cjs [hunk 8]
```diff
@@ -16,7 +16,7 @@ module.exports = {
       typescript: true,
     },
   },
-  ignorePatterns: ["node_modules/", "build/"],
+  ignorePatterns: ["node_modules/", "build/", "src/generatedNoCheck/"],
```
Comment (ericanderson) [inline monorepo/eslint-config-sane/library.cjs:19]: Dont let eslint touch the generated files. We want them pure for reviewing output.

File: package.json [hunk 9]
```diff
@@ -6,7 +6,7 @@
     "check-mrl": "mrl check --verbose",
     "ci:publish": "pnpm run prePublish && pnpm exec changeset publish",
     "ci:publishSnapshot": "pnpm run prePublish && pnpm exec changeset version --snapshot && pnpm exec changeset publish --no-git-tag --snapshot --tag=next",
-    "dev": "turbo run dev",
+    "dev": "turbo run dev --concurrency 20",
```
Comment (ericanderson) [inline package.json:9]: Make `pnpm dev` "just work"

File: packages/generator/src/WireOntologyDefinition.ts [hunk 10]
```diff
@@ -23,4 +23,5 @@ export interface WireOntologyDefinition {
   objectTypes: { [key: string]: gateway.components.ObjectTypeV2 };
   actionTypes: gateway.components.ActionTypeV2[];
   queryTypes: gateway.components.QueryTypeV2[];
+  linkTypes: Record<string, gateway.components.LinkTypeSideV2[]>;
```
Comment (ericanderson) [inline packages/generator/src/WireOntologyDefinition.ts:26]: I'm not committed to this format at all. I just wanted to get something in here.

File: packages/generator/src/v1.1/generateMetadataFile.ts [hunk 11]
```diff
@@ -32,7 +32,10 @@ export async function generateMetadataFile(
   import type { OntologyDefinition } from "@osdk/api";
   import type { Ontology as ClientOntology } from "@osdk/legacy-client";
   import type { Objects } from "./ontologyObjects";
-  ${objectNames.map((name) => `import {${name}} from "./objects/${name}";`)}
+  ${
+      objectNames.map((name) => `import {${name}} from "./objects/${name}";`)
+        .join("\n")
```
Comment (ericanderson) [inline packages/generator/src/v1.1/generateMetadataFile.ts:37]: We need to remember to do this. newline on join or we end up with "," that we dont want in the code

File: packages/generator/src/v1.1/wireObjectTypeV2ToV1ObjectInterfaceString.ts [hunk 12]
```diff
@@ -20,7 +20,7 @@ export function wireObjectTypeV2ToObjectInterfaceStringV1(
   input: ObjectTypeV2,
 ) {
   return `
-import { OntologyObject, LocalDate, TimeStamp, GeoShape, GeoPoint } from "@osdk/legacy-client";
+import { OntologyObject, LocalDate, Timestamp, GeoShape, GeoPoint } from "@osdk/legacy-client";
```
Comment (ericanderson) [inline packages/generator/src/v1.1/wireObjectTypeV2ToV1ObjectInterfaceString.ts:23]: Typo

File: turbo.json [hunk 13]
```diff
@@ -89,7 +89,7 @@
     },
 
     "check": {
-      "dependsOn": ["//#check-mrl", "lint", "build"]
+      "dependsOn": ["//#check-mrl", "lint", "build", "test"]
```
Comment (ericanderson) [inline turbo.json:92]: Make `turbo check` run tests

**Review by ericanderson — COMMENTED** [review]
Extra note

**Review by pashakondratyev — APPROVED** [review]

**Pull request merged**